src/Server.php

```diff
@@ -53,11 +53,11 @@ class Server
 
     /**
      * Create Server instance.
-     * @param FilesystemInterface $source The source file system.
-     * @param FilesystemInterface $cache  The cache file system.
-     * @param ApiInterface        $api    The image manipulation API.
+     * @param FilesystemInterface      $source The source file system.
+     * @param FilesystemInterface|null $cache  The cache file system.
+     * @param ApiInterface             $api    The image manipulation API.
      */
-    public function __construct(FilesystemInterface $source, FilesystemInterface $cache, ApiInterface $api)
+    public function __construct(FilesystemInterface $source, $cache, ApiInterface $api)
```

Comment:
I wonder if this is best. I really hate removing the `FilesystemInterface ` type-hinting, but I also don't want to change the parameter order because that would break backward compatibly.

We could do this:

```php
public function __construct(FilesystemInterface $source, FilesystemInterface $cache = null, ApiInterface $api = null)
```

Now clearly `ApiInterface` cannot be null, however if null is passed it will then get caught in the `setApi` method type-hinting. So, kind of weird, but maybe better than removing the `FilesystemInterface` type-hint?

Thoughts? /cc @GrahamCampbell